### ISSUE TITLE
Numpy dataset not recognising arrays?

### ISSUE THREAD

**Issue by Solin64** [deepchem/deepchem#2661]
Hi there

I am using some of my own formatted data on a CSV file to try and execute some protein-ligand modelling. I have the pdb_id, smiles and labels for my small molecules and fixed them without a hitch. When i got to:

dataset = dc.data.NumpyDataset(X=features, y=labels, ids=pdbids)
train_dataset, test_dataset = dc.splits.RandomSplitter().train_test_split(dataset)

I get an IndexError - arrays used as indices must be of integer (or boolean) type. The error is below:

Just wondering if I need to save my features and pdbids somewhere specific instead of the local file I am using?

Thanks for any advice!

![Protein modelling error](https://user-images.githubusercontent.com/88290302/129781499-66ef85fa-6ee7-428a-8f5d-1e8f341e2fcc.png)
